File: selenium-comparison/spec/tests/auth01_happy_path_spec.rb
require_relative '../spec_helper'

# Mirrors: tests/auth/auth-01-happy-path.spec.ts

RSpec.describe '3 — Authentication' do
  subject(:login) { LoginPage.new(@driver) }

  it 'AUTH-01 — Successful login with valid credentials redirects to home' do
    login.navigate_to_login

    expect(@driver.title).to eq('LOGIN')
    expect(login.email_input_visible?).to    be(true)
    expect(login.password_input_visible?).to be(true)
    expect(login.submit_button_visible?).to  be(true)

    login.login_as(
      ENV.fetch('TEST_USER_EMAIL'),
      ENV.fetch('TEST_USER_PASSWORD')
    )

    expect(@driver.current_url).to match(/\/home/)
    expect(@driver.title).to eq('Home')
    expect(login.logged_in_indicator_visible?).to be(true)
    expect(login.login_link_visible?).to be(false)
  end
end

File: tests/auth/auth-01-happy-path.spec.ts
// spec: specs/sanzu.plan.md
// seed: fixtures/fixtures.ts

import { test, expect } from '../../fixtures/fixtures';

test.describe('3 — Authentication', () => {
  test('AUTH-01 — Successful login with valid credentials redirects to home', async ({ loginPage, page }) => {
    // Step 2: Call loginPage.goto() to navigate to /login_desktop.
    await loginPage.goto();
    await expect(page).toHaveTitle('LOGIN');
    await expect(loginPage.loginForm.emailInput).toBeVisible();
    await expect(loginPage.loginForm.passwordInput).toBeVisible();
    await expect(loginPage.loginForm.submitButton).toBeVisible();

    // Step 3: Call loginPage.loginAs() with valid credentials.
    await loginPage.loginAs(process.env.TEST_USER_EMAIL!, process.env.TEST_USER_PASSWORD!);
    await expect(page).toHaveURL(/\/home/);
    await expect(page).toHaveTitle('Home');

    // Step 4: Assert authenticated username button is visible in the header nav list.
    await expect(loginPage.header.loggedInIndicator).toBeVisible();
    await expect(loginPage.header.loginLink).not.toBeVisible();
  });
});
